pkg/controller/elasticquota.go

```diff
@@ -0,0 +1,348 @@
+		klog.V(5).Infof("Elastic quota has been deleted %v", key)
+		return nil
+	}
+	if err != nil {
+		klog.V(3).Infof("Unable to retrieve elastic quota %v from store: %v", key, err)
+		return err
+	}
+
+	klog.V(5).Infof("Try to process elastic quota: %s", key)
+	used, err := ctrl.computeElasticQuotaUsed(namespace, eq)
+	if err != nil {
+		return err
+	}
+	// create a usage object that is based on the elastic quota version that will handle updates
+	// by default, we set used to the current status
+	usage := eq.DeepCopy()
+	usage.Status = schedv1alpha1.ElasticQuotaStatus{
+		Used: used,
+	}
+	// Ignore this loop if the usage value has not changed
+	if apiequality.Semantic.DeepEqual(usage.Status, eq.Status) {
+		return nil
+	}
+	if _, err = ctrl.schedClient.SchedulingV1alpha1().ElasticQuotas(namespace).
+		Update(context.TODO(), usage, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
+	ctrl.recorder.Event(eq, v1.EventTypeNormal, "Synced", fmt.Sprintf("Elastic Quota %s synced successfully", key))
+	return nil
+}
+
+func (ctrl *ElasticQuotaController) computeElasticQuotaUsed(namespace string, eq *schedv1alpha1.ElasticQuota) (v1.ResourceList, error) {
+	used := newZeroUsed(eq)
+	pods, err := ctrl.podLister.Pods(namespace).List(labels.NewSelector())
+	if err != nil {
+		return nil, err
+	}
+	for _, p := range pods {
+		if p.Status.Phase == v1.PodRunning {
+			used = quota.Add(used, computePodResourceRequest(p))
+		}
+	}
+	return used, nil
+}
+
+// eqAdded reacts to a ElasticQuota creation
+func (ctrl *ElasticQuotaController) eqAdded(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	klog.V(5).Infof("Enqueue new elastic quota key %v", key)
+	ctrl.eqQueue.AddRateLimited(key)
+}
+
+// eqUpdated reacts to a ElasticQuota update
+func (ctrl *ElasticQuotaController) eqUpdated(oldObj, newObj interface{}) {
+	ctrl.eqAdded(newObj)
+}
+
+// eqDeleted reacts to a ElasticQuota deletion
+func (ctrl *ElasticQuotaController) eqDeleted(obj interface{}) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	klog.V(5).Infof("Enqueue deleted elastic quota key %v", key)
+	ctrl.eqQueue.AddRateLimited(key)
+}
+
+// podAdded reacts to a pod creation
+func (ctrl *ElasticQuotaController) podAdded(obj interface{}) {
+	pod := obj.(*v1.Pod)
+	list, err := ctrl.eqLister.ElasticQuotas(pod.Namespace).List(labels.Everything())
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	if len(list) == 0 {
+		return
+	}
+	// todo(yuchen-sun) When elastic quota supports multiple instances in a namespace, modify this
+	ctrl.eqAdded(list[0])
+}
+
+// podUpdated reacts to a pod update
+func (ctrl *ElasticQuotaController) podUpdated(oldObj, newObj interface{}) {
+	oldPod := oldObj.(*v1.Pod)
+	newPod := newObj.(*v1.Pod)
+	if newPod.ResourceVersion == oldPod.ResourceVersion {
+		return
+	}
+	ctrl.podAdded(newObj)
```

Comment:
I guess you'd also need to verify its namespace is associated with an ElasticQuota, right? just like what `podAdded()` did above.

Reply:
I called podAdded on the last line, which will check what you said.

Reply:
SG.